numpy/core/tests/test_ufunc.py

```diff
@@ -327,7 +333,10 @@ def test_signature(self):
             pass
 
         # more complicated names for variables
-        assert_equal(umt.test_signature(2, 1, "(i1,i2),(J_1)->(_kAB)"), 1)
+        enabled, num_dims, ixs = umt.test_signature(2, 1, "(i1,i2),(J_1)->(_kAB)")
+        assert_equal(enabled, 1)
+        assert_equal(num_dims, (2, 1, 1))
+        assert_equal(ixs, (0, 1, 2, 3))
```

Comment:
Might be nicer to build a nested tuple here, and I don't envisage it being any harder